### DISCUSSION TITLE
`events` prop of nodes/edges to be deprecated

### DISCUSSION THREAD

**Discussion by harshaktg** [bcakmakoglu/vue-flow#1541] [Q&A] [answered]
I notice that events will be removed during node/edge config. So using the methods from the `useVueFlow` is the right way in the future? I'm curious on why did we make this change? Another question is - if we used different types of nodes and different click handlers for each of them, then should it now be combined into a single function and handle 3 logics within?

<img width="523" alt="image" src="https://github.com/user-attachments/assets/b85f0b27-1d7e-44cb-a60a-bfe1d50fabaa">

**Comment by harshaktg** [comment]
A follow up question on this. Can we not have a custom event like how we had before?

**Reply by bcakmakoglu** [reply, marked as answer]
> I notice that events will be removed during node/edge config.

Yes, in the next major version this object will not exist anymore.

> So using the methods from the useVueFlow is the right way in the future?

Either those or you can bind events directly to `<VueFlow>`, for example `<VueFlow @node-click="onNodeClick" />`

> I'm curious on why did we make this change?

Functions are not properly convertible into JSON, therefore any of the existing options that currently allow functions will be removed/replaced with something that is convertible to JSON.

>  Another question is - if we used different types of nodes and different click handlers for each of them, then should it now be combined into a single function and handle 3 logics within?

You are always able to bind your own click handlers in your own custom node component instead of using the "global" `onNodeClick` event (which is applied to a wrapper on nodes).

```vue
// CustomNode.vue
<template>
	<div @click="onClick">
		<div>{{ label }}</div>
	</div>
</template>
```

Otherwise, yes, you would have to include all logic inside a single click handler and differentiate the clicked node-type and execute the corresponding logic.

> Can we not have a custom event like how we had before?

You can have whatever event you want inside your custom node, VueFlow won't care about it.
"Custom" events right now don't do anything besides simply being forwarded to your custom node component, so there's no real difference whether this event was forwarded from a nodes object or was simply defined in the custom node itself.

**Reply by harshaktg** [reply]
Cool! Thank you @bcakmakoglu
